shared/settings/advanced/index.tsx

```diff
@@ -54,43 +56,83 @@ const UseNativeFrame = (props: Props) => {
   )
 }
 
-const Advanced = (props: Props) => {
-  const disabled = props.lockdownModeEnabled == null || props.hasRandomPW || props.settingLockdownMode
-  return (
-    <Kb.Box style={styles.advancedContainer}>
-      <Kb.Box style={styles.progressContainer}>
-        {props.settingLockdownMode && <Kb.ProgressIndicator />}
-      </Kb.Box>
-      <Kb.Box style={styles.checkboxContainer}>
-        <Kb.Checkbox
-          checked={props.hasRandomPW || !!props.lockdownModeEnabled}
-          disabled={disabled}
-          label={
-            'Forbid account changes from the website' +
-            (props.hasRandomPW ? ' (you need to set a password first)' : '')
-          }
-          onCheck={props.onChangeLockdownMode}
-          style={styles.checkbox}
-        />
-      </Kb.Box>
-      {!!props.setLockdownModeError && (
-        <Kb.Text type="BodySmall" style={styles.error}>
-          {props.setLockdownModeError}
-        </Kb.Text>
-      )}
-      {isLinux && <UseNativeFrame {...props} />}
-      {!Styles.isMobile && !isLinux && (
-        <Kb.Box style={styles.openAtLoginCheckboxContainer}>
+type AdvancedState = {
+  disableCertPinning: () => void | undefined
+  showCertPinningModal: boolean
+}
+
+class Advanced extends React.Component<Props, AdvancedState> {
+  constructor(props: Props) {
```

Comment:
you can just do
```
state = {
  disableCertPinning: undefined,
  showCertPinningModal: false,
}
``` 
and avoid the constructor

Reply:
Done.